### PR TITLE
feat: generate Docker image SBOM

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Use the `docker sbom` command to generate a CycloneDX
format BOM.

Optionally support logging into an AWS ECR first so that
the image can be pulled.

# Related
* cds-snc/platform-sre-security-support#94

### CODE REVIEW

**Review by mohdnr — CHANGES_REQUESTED** [review]
Do we want to deprecate access keys and instead have teams setup OIDC to this repo that has read access ?

**Comment by patheard** [comment]
> Do we want to deprecate access keys and instead have teams setup OIDC to this repo that has read access ?

I'm game to set that up - be a good exercise to get that rolled out across the projects

File: .github/workflows/generate_sbom.yml [hunk 1]
```diff
@@ -61,16 +70,16 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Configure AWS credentials
-        if: inputs.project_type == 'docker' && secrets.aws_access_key_id != '' && secrets.aws_secret_access_key != ''
+      - name: configure aws credentials using OIDC
+        if: inputs.project_type ='docker' && inputs.iam_role_to_assume != '' && inputs.iam_role_session_name != ''
```
Comment (patheard) [inline .github/workflows/generate_sbom.yml]: One tiny gotcha - we need the double `==` for the first check:
```
inputs.project_type == 'docker'
```

File: .github/workflows/generate_sbom.yml [hunk 2]
```diff
@@ -61,16 +70,16 @@ jobs:
...
-          aws-access-key-id: ${{ secrets.aws_access_key_id }}
-          aws-secret-access-key: ${{ secrets.aws_secret_access_key }}
-          aws-region: ca-central-1
+          role-to-assume: ${{ inputs.iam_role_to_assume }}
+          role-session-name: ${{ inputs.iam_role_session_name }}
+          aws-region: "ca-central-1"
 
       - name: Login to Elastic Container Registry
-        if: inputs.project_type == 'docker' && secrets.aws_access_key_id != '' && secrets.aws_secret_access_key != ''
+        if: inputs.project_type ='docker' && inputs.iam_role_to_assume != '' && inputs.iam_role_session_name != ''
```
Comment (patheard) [inline .github/workflows/generate_sbom.yml]: same double `==` gotcha

**Review by dinophile — APPROVED** [review]
Well it certainly looks good to me! :shipit:

**Review by mohdnr — APPROVED** [review]
real nice! 🚀 ✅

**Pull request merged**